### PR TITLE
Agent uninstall endpoint

### PR DESCRIPTION
**Pull request by RamosFe**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/24643|

## Description

Adds the `GET /agents/uninstall` endpoint and its related tests.

### CODE REVIEW

File: api/api/controllers/agent_controller.py [hunk 1]
```diff
@@ -1678,3 +1678,33 @@ async def get_agent_summary_os(pretty: bool = False, wait_for_complete: bool = F
...
+
+async def get_agent_uninstall_permission(pretty: bool = False, wait_for_complete: bool = False) -> ConnexionResponse:
+    """Check user's permission to uninstall agents
+
+    Parameters
+    ----------
+    pretty : bool
+        Show results in human-readable format
+    wait_for_complete : bool
+        Disable timeout response
```
Comment (GGP1) [inline api/api/controllers/agent_controller.py:1691]:
```suggestion
    pretty : bool
        Show results in human-readable format.
    wait_for_complete : bool
        Disable timeout response.
```

File: api/api/controllers/agent_controller.py [hunk 2]
```diff
@@ -1678,3 +1678,33 @@ async def get_agent_summary_os(pretty: bool = False, wait_for_complete: bool = F
...
+
+    Returns
+    -------
+    ConnexionResponse
+        API response.
+    """
+    f_kwargs = {}
+
+    dapi = DistributedAPI(f=agent.agent_check_uninstall_permission,
```
Comment (GGP1) [inline api/api/controllers/agent_controller.py:1700]:
```suggestion
    dapi = DistributedAPI(f=agent.check_uninstall_permission,
```

File: api/api/controllers/test/test_agent_controller.py [hunk 3]
```diff
@@ -1184,3 +1184,26 @@ async def test_get_agent_summary_os(mock_exc, mock_dapi, mock_remove, mock_dfunc
...
+@pytest.mark.parametrize("mock_request", ["agent_controller"], indirect=True)
+@patch('api.configuration.api_conf')
+@patch('api.controllers.agent_controller.DistributedAPI.distribute_function', return_value=AsyncMock())
+@patch('api.controllers.agent_controller.remove_nones_to_dict')
+@patch('api.controllers.agent_controller.DistributedAPI.__init__', return_value=None)
+@patch('api.controllers.agent_controller.raise_if_exc', return_value=CustomAffectedItems())
+async def test_get_agent_uninstall_permission(mock_exc, mock_dapi, mock_remove, mock_dfunc, mock_exp, mock_request):
+    """Verify 'get_agent_uninstall_permission' endpoint is working as expected."""
+    result = await get_agent_uninstall_permission()
+    mock_dapi.assert_called_once_with(f=agent.agent_check_uninstall_permission,
```
Comment (GGP1) [inline api/api/controllers/test/test_agent_controller.py:1199]:
```suggestion
    mock_dapi.assert_called_once_with(f=agent.check_uninstall_permission,
```

File: api/api/spec/spec.yaml [hunk 4]
```diff
@@ -8262,6 +8276,40 @@ paths:
...
+      responses:
+        '200':
+          description: "User has the necessary permissions to uninstall agents"
+          content:
+            application/json:
+              schema:
+                allOf:
+                - $ref: '#/components/schemas/ApiResponse'
+              example:
+                message: "Agent has permission to be uninstalled."
```
Comment (GGP1) [inline api/api/spec/spec.yaml:8300]:
```suggestion
                message: "User has permission to uninstall agents"
```

File: api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml [hunk 5]
```diff
@@ -1859,3 +1859,21 @@ stages:
...
+    request:
+      verify: False
+      url: "{protocol:s}://{host:s}:{port:d}/agents/uninstall"
+      method: GET
+      headers:
+        Authorization: "Bearer {test_login_token}"
+    response:
+      status_code: 200
+      json:
+        message: "Agent has permission to be uninstalled"
```
Comment (GGP1) [inline api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml:1878]:
```suggestion
        message: "User has permission to uninstall agents"
```

File: framework/wazuh/agent.py [hunk 6]
```diff
@@ -1513,3 +1513,19 @@ def get_full_overview() -> WazuhResult:
               'agent_version': stats_version, 'last_registered_agent': last_registered_agent}
 
     return WazuhResult({'data': result})
+
+
+@expose_resources(actions=["agent:uninstall"], resources=["*:*:*"], post_proc_func=None)
+def agent_check_uninstall_permission() -> WazuhResult:
```
Comment (GGP1) [inline framework/wazuh/agent.py:1519]:
```suggestion
def check_uninstall_permission() -> WazuhResult:
```

File: framework/wazuh/agent.py [hunk 7]
```diff
@@ -1513,3 +1513,19 @@ def get_full_overview() -> WazuhResult:
...
+    This function verifies whether the user has the necessary permissions
+    to uninstall agents by checking the assigned RBAC actions and resources.
+
+    Returns
+    -------
+    WazuhResult
+        WazuhResult object containing a message indicating if the
+        user has the permission to uninstall agents.
+    """
+    return WazuhResult({'message': "Agent has permission to be uninstalled"})
```
Comment (GGP1) [inline framework/wazuh/agent.py:1531]:
```suggestion
    return WazuhResult({'message': "User has permission to uninstall agents"})
```

File: framework/wazuh/tests/test_agent.py [hunk 8]
```diff
@@ -1593,3 +1593,11 @@ def test_unset_single_group_agent_ko(agent_basic_mock, group_exists_mock, get_gr
     with pytest.raises(WazuhException, match=f".* {expected_exc} .*"):
         Agent.unset_single_group_agent(agent_id, group_id, force=force)
 
+
+def test_agent_check_uninstall_permission():
```
Comment (GGP1) [inline framework/wazuh/tests/test_agent.py:1597]:
```suggestion
def test_check_uninstall_permission():
```

File: framework/wazuh/tests/test_agent.py [hunk 9]
```diff
@@ -1593,3 +1593,11 @@ def test_unset_single_group_agent_ko(agent_basic_mock, group_exists_mock, get_gr
...
+    """Check that agent_check_uninstall_permission returns the expected msg"""
+
+    result = agent_check_uninstall_permission()
```
Comment (GGP1) [inline framework/wazuh/tests/test_agent.py:1600]:
```suggestion
    result = check_uninstall_permission()
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

File: api/api/controllers/agent_controller.py (hunk 1 above)
Comment (RamosFe) [reply api/api/controllers/agent_controller.py]: Fixed in efb4a691b97c462325ee2b7e9c4faa616a435692.

File: api/api/controllers/agent_controller.py (hunk 2 above)
Comment (RamosFe) [reply api/api/controllers/agent_controller.py]: Fixed in efb4a691b97c462325ee2b7e9c4faa616a435692.

File: api/api/controllers/test/test_agent_controller.py (hunk 3 above)
Comment (RamosFe) [reply api/api/controllers/test/test_agent_controller.py]: Fixed in efb4a691b97c462325ee2b7e9c4faa616a435692.

File: api/api/spec/spec.yaml (hunk 4 above)
Comment (RamosFe) [reply api/api/spec/spec.yaml]: Fixed in efb4a691b97c462325ee2b7e9c4faa616a435692.

File: framework/wazuh/agent.py (hunk 6 above)
Comment (RamosFe) [reply framework/wazuh/agent.py]: Fixed in efb4a691b97c462325ee2b7e9c4faa616a435692.

File: framework/wazuh/agent.py (hunk 7 above)
Comment (RamosFe) [reply framework/wazuh/agent.py]: Fixed in efb4a691b97c462325ee2b7e9c4faa616a435692.

File: framework/wazuh/tests/test_agent.py (hunk 8 above)
Comment (RamosFe) [reply framework/wazuh/tests/test_agent.py]: Fixed in https://github.com/wazuh/wazuh/commit/efb4a691b97c462325ee2b7e9c4faa616a435692.

File: framework/wazuh/tests/test_agent.py (hunk 9 above)
Comment (RamosFe) [reply framework/wazuh/tests/test_agent.py]: Fixed in https://github.com/wazuh/wazuh/commit/efb4a691b97c462325ee2b7e9c4faa616a435692.

File: api/api/spec/spec.yaml [hunk 10]
```diff
@@ -8262,6 +8276,40 @@ paths:
...
+      responses:
+        '200':
+          description: "User has the necessary permissions to uninstall agents"
+          content:
+            application/json:
+              schema:
+                allOf:
+                - $ref: '#/components/schemas/ApiResponse'
+              example:
+                message: "User has permission to be uninstalled."
```
Comment (GGP1) [inline api/api/spec/spec.yaml:8300]:
```suggestion
                message: "User has permission to uninstall agents."
```

File: api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml [hunk 11]
```diff
@@ -1859,3 +1859,21 @@ stages:
...
+    request:
+      verify: False
+      url: "{protocol:s}://{host:s}:{port:d}/agents/uninstall"
+      method: GET
+      headers:
+        Authorization: "Bearer {test_login_token}"
+    response:
+      status_code: 200
+      json:
+        message: "User has permission to be uninstalled"
```
Comment (GGP1) [inline api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml:1878]:
```suggestion
        message: "User has permission to uninstall agents"
```

File: framework/wazuh/agent.py [hunk 12]
```diff
@@ -1513,3 +1513,19 @@ def get_full_overview() -> WazuhResult:
...
+    This function verifies whether the user has the necessary permissions
+    to uninstall agents by checking the assigned RBAC actions and resources.
+
+    Returns
+    -------
+    WazuhResult
+        WazuhResult object containing a message indicating if the
+        user has the permission to uninstall agents.
+    """
+    return WazuhResult({'message': "User has permission to be uninstalled"})
```
Comment (GGP1) [inline framework/wazuh/agent.py:1531]:
```suggestion
    return WazuhResult({'message': 'User has permission to uninstall agents'})
```

File: framework/wazuh/tests/test_agent.py [hunk 13]
```diff
@@ -1593,3 +1593,11 @@ def test_unset_single_group_agent_ko(agent_basic_mock, group_exists_mock, get_gr
     with pytest.raises(WazuhException, match=f".* {expected_exc} .*"):
         Agent.unset_single_group_agent(agent_id, group_id, force=force)
 
+
+def test_check_uninstall_permission():
+    """Check that agent_check_uninstall_permission returns the expected msg"""
+
+    result = check_uninstall_permission()
+    expected = WazuhResult({'message': "User has permission to be uninstalled"})
```
Comment (GGP1) [inline framework/wazuh/tests/test_agent.py:1601]:
```suggestion
    expected = WazuhResult({'message': 'User has permission to uninstall agents'})
```

**Review by GGP1 — CHANGES_REQUESTED** [review]

File: framework/wazuh/agent.py (hunk 12 above)
Comment (RamosFe) [reply framework/wazuh/agent.py]: Changed in b7dcc227e48732b77db2c92c1087e322a55d8ae9.

File: framework/wazuh/tests/test_agent.py (hunk 13 above)
Comment (RamosFe) [reply framework/wazuh/tests/test_agent.py]: Changed in b7dcc227e48732b77db2c92c1087e322a55d8ae9.

File: api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml (hunk 11 above)
Comment (RamosFe) [reply api/test/integration/test_rbac_white_agent_endpoints.tavern.yaml]: Changed in b7dcc227e48732b77db2c92c1087e322a55d8ae9.

File: api/api/spec/spec.yaml (hunk 10 above)
Comment (RamosFe) [reply api/api/spec/spec.yaml]: Changed in b7dcc227e48732b77db2c92c1087e322a55d8ae9.

**Review by GGP1 — APPROVED** [review]
LGTM

**Review by fdalmaup — APPROVED** [review]
GJ! LGTM

**Pull request merged**